### PR TITLE
[T2A2][W10-A4] Wang Zexin

### PR DESCRIPTION
**Pull request by wangzexin**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -262,14 +262,12 @@ private static void processProgramArgs(String[] args) {
...
-        if (args.length == 1) {
+        } else if (args.length == 1) {
             setupGivenFileForStorage(args[0]);
-        }
-
-        if(args.length == 0) {
+        } else if(args.length == 0) {
             setupDefaultFileForStorage();
+        } else {
+        	exitProgram();
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:271]: Thumbs up.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -156,6 +156,11 @@
      * Offset required to convert between 1-indexing and 0-indexing.COMMAND_
      */
     private static final int DISPLAYED_INDEX_OFFSET = 1;
+    
+    // Removed magic number in splitArgs
+    private static final int SPLIT_ARGS_INDEX1 = 0;
+    private static final int SPLIT_ARGS_INDEX2 = 1;
+    private static final int SPLIT_ARGS_INDEX3 = 2;
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:155]: Thumbs up for dealing with magic numbers.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -452,7 +454,8 @@ private static String getMessageForSuccessfulAddPerson(HashMap<PersonProperty, S
      */
     private static String executeFindPersons(String commandArgs) {
         final Set<String> keywords = extractKeywordsFromFindPersonArgs(commandArgs);
-        final ArrayList<HashMap<PersonProperty, String>> personsFound = getPersonsWithNameContainingAnyKeyword(keywords);
+        final ArrayList<HashMap<PersonProperty, String>> personsFound = 
+        		getPersonsWithNameContainingAnyKeyword(keywords);
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:462]: Thumbs up.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -803,7 +803,8 @@ private static void addPersonToAddressBook(HashMap<PersonProperty, String> perso
      * @param exactPerson the actual person inside the address book (exactPerson == the person to delete in the full list)
      * @return true if the given person was found and deleted in the model
      */
-    private static boolean deletePersonFromAddressBook(HashMap<PersonProperty, String> exactPerson) { // shouldn't this be starting with is?
+    private static boolean hasDeletePersonFromAddressBook(
+    		HashMap<PersonProperty, String> exactPerson) {
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:811]: Thumbs up for boolean naming.

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -388,7 +392,7 @@ private static String executeCommand(String userInputString) {
             return getMessageForInvalidCommandInput(
             		commandType, getUsageInfoForAllCommands());
         }
-    }
+	}
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:395]: Thumbs up for SLAP.

File: src/seedu/addressbook/AddressBook.java [hunk 6]
```diff
@@ -1170,8 +1170,12 @@ private static String getUsageInfoForExitCommand() {
...
      * @return  string without the sign
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefixSign(String fullString, String prefix) {
+    	if (fullString.startsWith(prefix)) {
+    		return fullString.replaceFirst(prefix, "");
+    	} else {
+    		return fullString;
+    	}
```
Comment (hwkchia) [inline src/seedu/addressbook/AddressBook.java:1178]: A bug thumbs up!

**Review by hwkchia — COMMENTED** [review]
Good job.  As mentioned during the class, just be careful which branch you are committing to; dealing with the hash map shouldn't be here.

**Comment by hwkchia** [comment]
@wangzexin Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/AddressBook.java (hunk 2 above)
Comment (wangzexin) [reply src/seedu/addressbook/AddressBook.java:155]: Thank you!

File: src/seedu/addressbook/AddressBook.java (hunk 6 above)
Comment (wangzexin) [reply src/seedu/addressbook/AddressBook.java:1178]: A bug :(

File: src/seedu/addressbook/AddressBook.java (hunk 1 above)
Comment (wangzexin) [reply src/seedu/addressbook/AddressBook.java:271]: Thank you!

**Pull request closed without merging**